src/policy_artifacthub.rs

```diff
@@ -577,8 +571,10 @@ mod tests {
             None,
         );
         assert_eq!(
-            arthub.unwrap_err().to_string(),
-            "policy version must be in semver: unexpected character 'n' while parsing major version number"
+            arthub.unwrap_err(),
+            ArtifactHubError::NoSemverVersion(String::from(
```

Comment:
I'm fine with it, but there's still the chance to have the test break because we change something inside of the error string.

I would do something like that:

```rust
assert!(matches!(arthub.unwrap_err(), ArtifactHubError::NoSemverVersion))
```

I think checking for the error type is good enough.

There are also other error checks below that look after specific strings, I don't know if it would make sense to change them too

Reply:
True, better. Done in https://github.com/kubewarden/policy-evaluator/pull/246/commits/4338dd4866ab508a0b90d3180f1f7aaa38f7420f.